_posts/2024-02-21-securing-wildfly-apps-okta-openshift.adoc

```diff
@@ -0,0 +1,260 @@
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
+```
+
+<1> *wildfly-maven-plugin* provisions a WildFly server with the specified layers with our application deployed.
+<2> *elytron-oidc-client* automatically adds the native OIDC client subsystem to our WildFly installation.
+
+. Examine the https://github.com/PrarthonaPaul/elytron-examples/blob/simple-webapp-okta/simple-webapp-okta/src/main/webapp/WEB-INF/web.xml[web.xml] file.
+```
+... <login-config> <auth-method>OIDC</auth-method> (1) </login-config> ...
+```
+When *elytron-oidc-client* subsystem sees *auth-method* is set to *OIDC*, it enables OIDC authentication mechanism for the application.
+
+. Examine the https://github.com/PrarthonaPaul/elytron-examples/blob/simple-webapp-okta/simple-webapp-okta/src/main/webapp/WEB-INF/oidc.json[oidc.json] file. The oidc.json is used to configure the native OIDC client subsystem.
+```
+{
+    "client-id" : "${env.CLIENT_ID}",
+    "provider-url" : "${env.OIDC_PROVIDER_URL}",
+    "public-client" : "false",
+    "principal-attribute" : "IDTClaim",
+    "ssl-required" : "EXTERNAL",
+    "credentials" : {
+        "secret" : "${env.CLIENT_SECRET}"
+    }
+}
+```
+* Client_ID is the unique identifier for our client used by the Okta OpenID provider. Usually for Keycloak, you can create your own client, but for Okta, you are assigned an id. As a result, we are using environment variables to specify this. We will set the value in 
+* The provider URL, which is the URL for the authorization server that we created, is specified as an environment variable. We will set its value in the helm configuration.
+* When public-client set to false, client credentials are sent when communicating with the OpenID provider.
+* We specify that the user name of the identity, which in our case is alice, is to be used as the principal for the identity. We are extracting this information here using a custom claim in the ID token.
+* When ssl-required is set to EXTERNAL, only the communication with external clients happens over HTTPs
+
+== Get the Application URL
+
+Once the WildFly server has been provisioned, use the following command to find the URL for your example application:
+
+```
+SIMPLE_WEBAPP_OKTA_URL=https://$(oc get route oidc-app --template='{{ .spec.host }}') &&
+echo "" &&
+echo "Application URL: $SIMPLE_WEBAPP_OKTA_URL/simple-webapp-okta"  &&
+echo "Valid redirect URI: $SIMPLE_WEBAPP_OKTA_URL/simple-webapp-okta/secured/" &&
+echo ""
+```
+We’ll make use of these URLs in the next two sections.
+
+== Finish Configuring Okta
+
+Go back to the _General Settings_ for your application and click on _Edit_.  add the `Valid redirect URI` under the `Sign-in redirect URIs` and check the `Allow wildcard * in login URI redirect` field and hit Save.
```

Comment:
Would be good to make General Settings and Edit bold instead

Reply:
fixed!